crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -226,9 +226,8 @@ pub(crate) fn type_check_expression(
 
             block_type
         }
-        HirExpression::Prefix(_) => {
-            // type_of(prefix_expr) == type_of(rhs_expression)
-            todo!("prefix expressions have not been implemented yet")
+        HirExpression::Prefix(prefix_expr) => {
+            type_check_expression(interner, &prefix_expr.rhs, errors)
```

Comment:
This is the only change I can find that would relate to a bitwise not. What is the plan for `!`? We want it now to work for booleans and as a bitwise operator like we do for `&` and `|`?

While we're fixing this code can we also get proper type checking for prefix operators? We should have a new function which checks if the rhs is numeric for `-` or if it is numeric or a boolean for `!`. Both should return the same type as their operand as well.

Reply:
Yes, `!` will be the bitwise not